pkg/frontend/querymiddleware/split_and_cache.go

```diff
@@ -139,7 +154,7 @@ func (s *splitAndCacheMiddleware) Do(ctx context.Context, req Request) (Response
 
 		for _, splitReq := range splitReqs {
 			// Do not try to pick response from cache at all if the request is not cachable.
-			if !isRequestCachable(splitReq.orig, maxCacheTime, s.cacheUnalignedRequests, s.logger) {
+			if !isRequestCachable(splitReq.orig, maxCacheTime, s.cacheUnalignedRequests, s.logger, s.metrics.notCachableReason) {
```

Comment:
`isRequestCachable()` is called for each split request. This means that if you run a query spanning over 2 days, we track it twice (and for each day the result of `isRequestCacheble()` may be different). I think it's fine (I mean, what we want) but I would rather clarify it in the metric description that it's tracked for each split query when the splitting by interval is enabled.